gradio/components/checkboxgroup.py

```diff
@@ -82,6 +85,9 @@ def __init__(
                 f"Invalid value for parameter `type`: {type}. Please choose from one of: {valid_types}"
             )
         self.type = type
+        if show_select_all and show_label is False:
+            warnings.warn("`show_select_all` requires `show_label` to be True")
```

Comment:
maybe instead, the select all checkbox is still shown in this case, but with no label.